### PR TITLE
support various compression formats used by EPICS area detector

### LINKED ISSUE #199: hdf5plugin package supports compression formats
As [advised](https://docs.h5py.org/en/stable/high/dataset.html?highlight=blosc#custom-compression-filters), use the [`hdf5plugin`](http://www.silx.org/doc/hdf5plugin/latest/index.html) package to support various compression formats used by EPICS area detector, such as [BLOSC](https://www.blosc.org/).

### PR DESCRIPTION
**Pull request by prjemian**
- close #199

### CODE REVIEW

**Comment by prjemian** [comment]
apply advice from https://stackoverflow.com/questions/47359528

**Comment by prjemian** [comment]
Accept that Codacy will report `Unused import hdf5plugin`.  On `import hdf5plugin`, data compression codecs are installed and used by h5py calls from within the module.

Don't want to silence that report from flake8 configuration since we still want to detect other unused `import` statements.

**Comment by prjemian** [comment]
@carterbox Ready for review!

File: punx/tests/test_issue133.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ def test_environment_restriction():
 
         # test that _some_ restriction exists
         assert len(item) > len("h5py")
-        req = item[len("h5py"):]
+        req = item[len("h5py"):].strip()
```
Comment (carterbox) [inline punx/tests/test_issue133.py:51]: What does this do?

**Review by carterbox — CHANGES_REQUESTED** [review]

File: punx/tests/test_issue133.py (hunk 1 above)
Comment (prjemian) [reply punx/tests/test_issue133.py:51]: The `strip` allows for the [optional whitespace](https://github.com/prjemian/punx/commit/1e44450769a47d23124b4024534497832c82a2b0) before the pinned requirement: `h5py >=3`

**Comment by prjemian** [comment]
Thanks for the quick review!  I recycled for re-examination.

**Review by carterbox — APPROVED** [review]

**Comment by prjemian** [comment]
@carterbox Thanks!  The point this covers was tracked back to a wrong version of h5py installed.

**Comment by prjemian** [comment]
And the (new) strip removes the optional whitespace.

**Comment by prjemian** [comment]
It checks that h5py has a revision pin such as `>=1.23`

**Pull request merged**